public/graph.js

```diff
@@ -730,15 +718,181 @@ class GraphController {
                 .text("Total Outgoing Message(s) / day");
         }
 
+        function drawOneDayFailedGraph(yLimitFailed) {
+            // Set Y axis limit to max of daily values or to the value inputted by the user
+            let yLimitFailedTotal = d3.max(dailyFailedTotal, d => d.total_errored);
+
+            if (isYLimitFailedManuallySet != true) {
+                yLimitFailed = yLimitFailedTotal;
+            }
+
+            // set scale domain for failed graph
+            y_total_failed_sms.domain([0, yLimitFailed]);
+            failed_messages_x_axis_range.domain(d3.extent(data, d => new Date(d.datetime)));
+
+            d3.selectAll(".redrawElementFailed").remove();
+            d3.selectAll("#failedLine10min").remove();
+            d3.selectAll("#failedLine1day").remove();
+
+            // Add the X Axis for the total failed sms graph
+            total_failed_sms_graph
+                .append("g")
+                .attr("class", "redrawElementFailed")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(failed_messages_x_axis_range)
+                    .ticks(d3.timeDay.every(1))
+                    .tickFormat(dayDateFormatWithWeekdayName))
+                // Rotate axis labels
+                .selectAll("text")
+                .style("text-anchor", "end")
+                .attr("dx", "-.8em")
+                .attr("dy", ".15em")
+                .attr("transform", "rotate(-65)");
+
+            // Add X axis label for the total failed sms graph
+            total_failed_sms_graph
+                .append("text")
+                .attr("class", "redrawElementFailed")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Time (D:H:M:S)");
+            
+            // Add the Y Axis for the total failed sms graph
```

Comment:
The order for the other graphs is add Y axis then X axis rearrange this for code readability

Reply:
Let me change when I'll be creating bar graphs